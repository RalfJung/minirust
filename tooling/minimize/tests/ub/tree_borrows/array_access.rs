extern crate intrinsics;
use intrinsics::*;

//@ compile-flags: --minimize-tree-borrows
fn main() {
    unsafe {
        let mut data = [0u8, 1, 2, 3];
        let x = &mut data[0] as *mut u8; // (x, [R, R, R, R])
        let y = &mut *x; // (x, [R, R, R, R]) -> (y, [R, R, R, R])
        let yraw = y as *mut u8;

        *x.add(1) = 42; // (x, [R, A, R, R]) -> (y, [R, D, R, R])

        print(*yraw.add(0));
        print(*yraw.add(2));
        print(*yraw.add(3));

        *yraw.add(1) = 57;
    }
}
