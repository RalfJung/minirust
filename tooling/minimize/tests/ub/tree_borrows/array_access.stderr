fatal error: UB: Tree Borrows: Child writing a pointer with the Disabled permission
